src/main/java/core/basesyntax/strategy/OperationStrategy.java

```diff
@@ -0,0 +1,5 @@
+package core.basesyntax.strategy;
+
+public interface OperationStrategy {
```

Comment:
The idea of **Strategy** pattern  to have method  `OperationHandler get(Operation type)` that takes type of operation and return the class Handler (or Processor). The OperationHandler should have children classes BalanceHadler, PurchaseHandler etc. with method `perform()` that will do calculations and save the result to `Storage fruitMap.` You should refactor the code with correct usage of **Strategy** pattern.
Example:
```
public class OperationStrategy {
    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap;

    public OperationStrategy(Map<FruitTransaction.Operation,
            OperationHandler> operationHandlerMap) {
        this.operationHandlerMap = operationHandlerMap;
        operationHandlerMap.put(FruitTransaction.Operation.BALANCE, new BalanceOperation());
        operationHandlerMap.put(FruitTransaction.Operation.SUPPLY, new SupplyHandler());
        operationHandlerMap.put(FruitTransaction.Operation.PURCHASE, new PurchaseHandler());
        operationHandlerMap.put(FruitTransaction.Operation.RETURN, new ReturnOperation());
    }

    public OperationHandler get(FruitTransaction.Operation type) {
        return operationHandlerMap.get(type);
    }
}
```

Example of Handler
```
public class PurchaseHandler implements OperationHandler {
    @Override
    public void perform(FruitTransaction transaction, Storage storage) {
        validateOperation();
        storage.put(transaction.getFruit(), oldQuantity - transaction.getQuantity());
    }
}
```